### PR TITLE
constification

### PR DESCRIPTION
**Pull request by joseluis**
This is a bigger update. Some notes:

- Made constant the from_le_bytes and as_le bytes public methods.  
- Changed the return value of as_le_bytes from a slice to an array. That'd be a breaking change, but I think is more consistent like this.
- Added corresponding big-endian methods.
- Added most constants from std::f64, except MIN_10_EXP, MAX_10_EXP, MIN_EXP, MAX_EXP.
- added tests.

I also have found some things I dont completely understand, but I'm gonna open an issue for that.

### CODE REVIEW

**Comment by joseluis** [comment]
I don't understand why the indicated machine epsilon limit is 1 order of magnitude less than the current enforced limit, which seemingly works well.

I've made thousands of iterations with both 10e-15 and 10e-16 Decimal64 values, which is the smallest decimal type implementing arithmetic operations, and it seems to work the well in both cases, without stepping over intermediate values...

Should we change EPSILON to one more order of magnited of precision even if it disagrees with wikipedia? I didn't find anything in the spec about that...

File: src/bitstring/fixed64.rs [hunk 1]
```diff
@@ -13,25 +13,286 @@ A 64bit decimal number.
...
+        Bitstring64(FixedBinaryBuf::from_le_bytes([0, 0, 0, 0, 0, 0, 0, 120]));
+
+    /// Negative infinity (−∞).
+    pub const NEG_INFINITY: Self =
+        Bitstring64(FixedBinaryBuf::from_le_bytes([0, 0, 0, 0, 0, 0, 0, 248]));
+}
+
+#[test]
+#[cfg(test)]
+fn consts_128() {
```
Comment (KodrAus) [inline src/bitstring/fixed64.rs:178]:
```suggestion
fn consts_64() {
```

File: src/bitstring/fixed64.rs [hunk 2]
```diff
@@ -13,25 +13,286 @@ A 64bit decimal number.
...
+    /// Smallest positive normal value.
+    pub const MIN_POSITIVE: Self =
+        Bitstring64(FixedBinaryBuf::from_le_bytes([1, 0, 0, 0, 0, 0, 0, 0]));
+
+    /// Largest finite value.
+    pub const MAX: Self = Bitstring64(FixedBinaryBuf::from_le_bytes([
+        255, 252, 243, 207, 63, 255, 248, 119,
+    ]));
+
+    /// Not a Number (NaN).
```
Comment (KodrAus) [inline src/bitstring/fixed64.rs:164]: Maybe we could call this out as _Not a Number, with a zero payload_? One of the goals of this project is to take advantage of the fact that NaN payloads are properly specified for decimal numbers, so we can safely transport them.

**Review by KodrAus — COMMENTED** [review]
> Changed the return value of as_le_bytes from a slice to an array.

This seems good to me :+1: When we have the knowledge that the array is of a particular size it's good to pass that along.

> I don't understand why the indicated machine epsilon limit is 1 order of magnitude less than the current enforced limit, which seemingly works well.

Hmm, I'm not too sure either yet to be honest. Maybe we keep at as-is for now and open an issue for it? It's probably least surprising to stay consistent with what other libraries consider epsilon to be?

File: src/bitstring/fixed64.rs (hunk 2 above)
Comment (joseluis) [reply src/bitstring/fixed64.rs:164]: Sure, that'd be better.

**Pull request merged**